var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -489,56 +530,20 @@ def setup_build_environment(self, env):
     # likely to be up to date in supporting *all* of MFEM's
     # configuration options. So, don't use CMake
     #
-    def configure(self, spec, prefix):
+    def get_make_config_options(self, spec, prefix):
         def yes_no(varstr):
             return "YES" if varstr in self.spec else "NO"
 
-        # See also find_system_libraries in lib/spack/llnl/util/filesystem.py
-        # where the same list of paths is used.
-        sys_lib_paths = [
-            "/lib64",
-            "/lib",
-            "/usr/lib64",
-            "/usr/lib",
-            "/usr/local/lib64",
-            "/usr/local/lib",
-            "/usr/lib/x86_64-linux-gnu",
-        ]
-
-        def is_sys_lib_path(dir):
-            return dir in sys_lib_paths
-
-        xcompiler = ""
-        xlinker = "-Wl,"
-        if "+cuda" in spec:
-            xcompiler = "-Xcompiler="
-            xlinker = "-Xlinker="
-        cuda_arch = None if "~cuda" in spec else spec.variants["cuda_arch"].value
+        xcompiler = "" if "~cuda" in spec else "-Xcompiler="
 
         # We need to add rpaths explicitly to allow proper export of link flags
         # from within MFEM.
 
-        # Similar to spec[pkg].libs.ld_flags but prepends rpath flags too.
-        # Also does not add system library paths as defined by 'sys_lib_paths'
-        # above -- this is done to avoid issues like this:
-        # https://github.com/mfem/mfem/issues/1088.
         def ld_flags_from_library_list(libs_list):
```

Comment:
(minor) it looks like these definitions could be removed entirely (and whatever calls them could directly call `self.ld_flags_from_library_list`).

Reply:
I specifically added these local functions, so that I didn't need to add the `self.` in front of all the calls we already had. Is there another way to avoid adding the `self.`?

Reply:
You could say `ld_flags_from_library_list = self. ld_flags_from_library_list` (assign it like a variable), which would be a bit clearer on what the intent is.